### DISCUSSION TITLE
Multiple readers for multiple simultaneous copies

### DISCUSSION THREAD

**Discussion by Jovuf** [enzo1982/freac#321] [Help & Support]
Hi!
I'm looking to copy about 1500 cds from a media library, so I want to get 4 external usb cd players in addition to the one that's built into my laptop, which will make me 5 in all.
I've noticed that in a 7 hour day I can rip about 40 cd's just with my laptop's drive.
I calculated that with 5 readers I could copy 1500 in 1 week...

My config is win 10, intel i7, 4gb ram

It is possible to assign the output directory, is it possible to assign the input directory for each window duplicate?
1 for (D:)
2 for (F:)
3 for (G:)
4 for (H:)
5 for (I:)
Even if it means restarting a copying configuration every day I start copying CDs.

Day 1
I just bought a second player to try out...
the program, apparently, will take into account in the same list the copy

I then have to manually dissociate the different playlists... Unfortunately, my copying time has increased.
For an average of 7 mins to 15 mins usually, I got a first result of 1 hour for one and 7 mins for the other, so I tweaked a bit, there I was able to be at 20 mins for the two cds in parrallel ...
I erased the data and started over with the same two CDs, I'm at 1 hour for one and 40 minutes for the other

How to optimize all this please
or else if you have other sources I am interested

Thank you :D ^^

**Comment by enzo1982** [comment]
Hi,

fre:ac has integrated support for ripping simultaneously with multiple drives, so it's not necessary and in fact not recommended to start multiple instances of fre:ac to achieve this.

To rip using multiple drives, there are basically two options:

1. Load all drives, then rip, change the CDs and repeat.
Here you would put a disc in each drive, then wait for each disc to be loaded into the joblist and then start the ripping process with the tracks from all drives. fre:ac should rip from all drives in parallel.
In this mode, it's handy to enable the auto-load and auto-eject features in the ripper settings, but leave the auto-rip option disabled.

2. Fully automated ripping, just change the CDs as they are finished.
For this you would enable all three automatization options. This makes fre:ac start a ripping job automatically as soon as tracks are loaded into the joblist. Multiple ripping jobs will run in parallel, so as soon as one disc is finished and gets ejected, you just put in the next disc and close the drive tray and fre:ac should automatically start ripping that disc.

Some notes:

- On Windows, the auto-load feature can be a bit fuzzy at times. Please try if it works on your system and with your drives. Generally, if it works it should always work, but there are some systems where Windows just never sends notifications about the arrival of new discs.
- There is no option to assign different output directories to different drives, but as fre:ac creates a separate folder for each album by default, this should not be a problem as long as title data is available for your discs in freedb/GnuDb. This applies for the default output filename pattern of `<albumartist> - <album>\<track> - <artist> - <title>`. If you want to make absolutely sure that each disc gets put into its own folder, you can add the `<discid>` placeholder to the output filename pattern, e.g. like this: `<discid>\<track> - <artist> - <title>` You can find more information about the available output filename placeholders [here](https://freac.org/manual/en/howto.html#anchor_pat).
- When there is no information available for a disc in the freedb/GnuDb database, fre:ac will not attempt to start ripping automatically even if that option is enabled. This is to give you a chance to enter title information manually before ripping.
- It may be useful to know that you can clear the joblist as soon as a ripping job is started. The running job will still rip all the tracks that were present in the joblist when it was started. This allows to start rips from multiple drives manually if needed: Load drive 1 into the joblist, check the title information, start the ripping process, clear the joblist and immediately continue with drive 2 while drive 1 is ripping in the background.

Regarding playlists, with the default settings, fre:ac will create a separate playlist for each album (this is determined by the playlist file naming pattern which is `<albumartist> - <album>\<albumartist> - <album>` by default). If you are getting a single playlist with tracks from different CDs, make sure the _"Create only a single playlist and/or cue sheet file per conversion"_ is disabled in the playlist settings.

With the default settings, you should get one folder per album and each folder should contain its own playlist file with the tracks of that disc.

Hope this will be helpful.

**Comment by enzo1982** [comment]
Wondering why your rips are so slow, though. Did you limit the ripping speed in the settings? Or do you have cdparanoia enabled?

I recommend enabling cdparanoia only as a last resort as it can be very slow and is usually not necessary with modern drives (i.e. any drive that was built in the past ~20 years). fre:ac will check rips against the AccurateRip database by default and if that yields a successful verification, there is no need for cdparanoia at all. When a verification fails, fre:ac will display an error message after the rip, so you can try the failed tracks again.

With fre:ac's default settings, I can usually rip a CD in about 2.5 to 3 minutes.

**Comment by Jovuf** [comment]
Hi, I definitely can't copy several cds at the same time, I have 5 cd players and when I insert my cds they all end up in the same list.
What interests me is to copy them all at the same time and in parallel.
I can't find the option "create a single playlist and/or cue sheet file per conversion" but I don't know where it is.

Also I found this option but it doesn't seem to be assigned anything:

configure general settings > Resource > parallel processing > enable parallel processing or enable super fast processing

But I don't get parallel processing :/

**Reply by Lee-Carre** [reply]
Duplicate of #334.

Possibly intentional.

Let's imagine you have a set/batch of 10 tracks to encode. It's better to have 100% of 2 tracks (with 0% of the remaining 8), than have all 10 be partially complete.
Compare downloading files.

It is likely more efficient (based on how encoders work) to have more resources focused on a few at a time, than trying to maximise the count of concurrent track encodings. Context switching has a cost. More data to keep in memory, simultaneously (which, unless you have a vast amount, will slow down the techniques used to find the optimal compression (FLAC's best compsession preset uses an exhaustive (brute-force) search of the problem space).

Besides memory capacity, there's also I/O limitations. On modern hardware, you'll likely be able to encode faster than you can read from a CD, or write files to storage (HDD, or whatever).

---

You also can't start listening until something reaches 100%.

---

So, other than ‘bigger numbers’, why exactly do you want a high-count of concurrent encodings, as opposed to somewhat more serial (up to the limits of your hardware)?

**Comment by stevednm** [comment]
I have three CD drives (1 internal and 2 external), and use all three at the same time for simultaneous, parallel ripping/encoding.
Enzo's answers above are very clear, precise and helpful.

I load each CD drive's contents into the Freac joblist, edit the album name of each in the Tags-->Albums tab(s), and in the Options menu-->Encoders, I precede the Output filenames-->Filename pattern with <album> followed by a slash (backslash on Windows, forward slash in Linux). I use Linux, so the Filename pattern in Linux begins: <album>/
Then I click the Start-Encoding button (or Encode menu-->Start Encoding, and Freac creates the directories in the specified output folder (Joblist-->Output folder), using the (sometimes edited) Album name for each of the disks.
So although the tracks from all 3 CDs are in the same joblist, the tracks for each of the CDs are ripped (copied and encoded) and placed into their own separate directory in the output folder.

**Comment by stevednm** [comment]
In my response above, two words enclosed in less-than and greater-than symbols were removed (I guess they aren't normally allowed in text responses, so I'll try to use the insert-code button in the reply editor instead--
I precede the Output filenames-->Filename pattern with `<album>` followed by a slash (backslash on Windows, forward slash in Linux). I use Linux, so the Filename pattern begins with `<album>/` (`<album>\` in Windows)
